### DISCUSSION TITLE
Documentation on guitar pro file formats

### DISCUSSION THREAD

**Discussion by raul4247** [CoderLine/alphaTab#464] [Q&A] [answered]
# Question

Hello There! I'm working in a personal project and using alphaTab to display tabs from GuitarPro files. Is there any documentation on how the GuitarPro files work in more detail? How the alphaTab devs were able to read it's format?

**Comment by Danielku15** [comment, marked as answer]
1. For Guitar Pro 4 there was once a documentation shared from Arobas: http://dguitar.sourceforge.net/GP4format.html from this more details could be derived on further details. But most details of Guitar Pro 3-5 is learned via TuxGuitar and MuseScore. The rest was more with creating test files and checking which bytes changed. 
2. Guitar Pro 6 only had a special container format which was shared in some forums by the developers at Arobas. Having the container decryption working, there is just a XML file inside which can be read. 
3. Guitar Pro 7 then became a Zip with the same XML file inside. 

From there it is a pure "reading/learning/trial-and-error" approach to find out how it works. But overall there is no full documentation on everything.
